### PR TITLE
Clear out all OAuth token on auth config change

### PR DESCRIPTION
**Pull request by bdpiparva**
`OauthTokenSweeper` is a config change listner. Which clears out OAuth
tokens on security config change. This class is using SecurityConfig#hasSecurityMethodChanged
to determine change in security config

Updated `SecurityConfig#hasSecurityMethodChanged` to check config change
in auth config along with inbuilt password file and ldap

### CODE REVIEW

File: config/config-api/src/com/thoughtworks/go/config/SecurityConfig.java [hunk 1]
```diff
@@ -126,9 +126,12 @@ public boolean hasSecurityMethodChanged(SecurityConfig newSecurity) {
         if (newSecurity == null) {
             return true;
         }
+
         boolean ldapChanged = !ObjectUtil.equal(ldapConfig, newSecurity.ldapConfig());
         boolean passwordFileChanged = !ObjectUtil.equal(passwordFileConfig, newSecurity.passwordFileConfig());
-        return ldapChanged || passwordFileChanged;
+        final boolean securityAuthConfigChanged = !ObjectUtil.equal(securityAuthConfigs, newSecurity.securityAuthConfigs());
+
+        return ldapChanged || passwordFileChanged || securityAuthConfigChanged;
```
Comment (maheshp) [inline config/config-api/src/com/thoughtworks/go/config/SecurityConfig.java]: I think any changes to security config should not affect oAuth, so wondering if we need to clean the tokens at if there is a change in security. But, we need to give a thought bout the expected behaviour when security is enabled and disabled.

**Comment by jyotisingh** [comment]
@bdpiparva - closing this one. We agreed on not clearing off the tokens when the config changes. With the support for gadgets gone, we shouldn't even need this.

**Comment by ketan** [comment]
> @bdpiparva - closing this one. We agreed on not clearing off the tokens when the config changes. With the support for gadgets gone, we shouldn't even need this.

I thought that the understanding was that we remove the OAuthTokenSweeper altogether?

**Comment by jyotisingh** [comment]
@ketan  - yes, thats what I meant by not clearing off the tokens. That will be a separate PR I guess.

**Pull request closed without merging**